app/src/main/java/org/jellyfin/mobile/player/ui/PlayerFragment.kt

```diff
@@ -262,30 +264,34 @@ class PlayerFragment : Fragment() {
     fun onFastForward() = viewModel.fastForward()
 
     /**
-     * @return true if the audio track was changed
+     * @param callback called if track selection was successful and UI needs to be updated
      */
-    fun onAudioTrackSelected(index: Int): Boolean {
-        return viewModel.selectAudioTrack(index)
+    fun onAudioTrackSelected(index: Int, callback: TrackSelectionCallback): Job = lifecycleScope.launch {
+        if (viewModel.selectAudioTrack(index)) {
+            callback.onTrackSelected(true)
```

Comment:
Why not make this suspended instead of using a callback?

Reply:
Same question for the other methods

Reply:
I didn't want to add coroutines to the UI part

Reply:
We can use the viewmodel lifecycle so I don't really see the issue.

Reply:
Need to check if PlayerMenus has that. But it's an option, true.